geniusyield-market-maker/src/GeniusYield/MarketMaker/Utils.hs

```diff
@@ -53,3 +66,132 @@ instance HasPartialOrderNftScript DEXInfo where
 
 instance HasPartialOrderConfigAddr DEXInfo where
   getPartialOrderConfigAddr = dexPartialOrderConfigAddr
+
+
+-------------------------------------------------------------------------------
+-- Common Price Resolution
+-------------------------------------------------------------------------------
+
+-- | Generic time resolution for OHLC Candles
+data CommonResolution = CRes5m | CRes15m | CRes30m | CRes1h | CRes4h
+                      | CRes1d | CRes1w  | CRes1mo
+                      deriving stock (Eq, Ord, Show, Generic)
+
+instance FromJSON CommonResolution
+
+class PriceResolution a where
+  resolutionInherited :: Map.Map CommonResolution a
+
+instance PriceResolution Maestro.Resolution where
+  resolutionInherited = Map.fromList [ (CRes5m, Res5m), (CRes15m, Res15m), (CRes30m, Res30m), (CRes1h, Res1h)
+                                     , (CRes4h, Res4h), (CRes1d, Res1d), (CRes1w, Res1w), (CRes1mo, Res1mo) ]
+
+fromCommonResolution :: PriceResolution a => CommonResolution -> IO a
+fromCommonResolution = maybe (throwIO $ userError "Undefined common price resolution.") pure
+                       . flip Map.lookup resolutionInherited
+
+
+-------------------------------------------------------------------------------
+-- TapTools OHLCV query
+-------------------------------------------------------------------------------
+
+-- | Taptools time resolution for OHLC Candles
+data TtResolution = TtRes3m | TtRes5m | TtRes15m | TtRes30m | TtRes1h | TtRes2h | TtRes4h | TtRes12h
```

Comment:
Likewise, remove `TtRes` prefix for json instance.

Reply:
Done for `FromJSON` instance.  Will also implement `ToJSON` instance.